### PR TITLE
Add Gradle Enterprise exceptions log filter

### PR DESCRIPTION
**Pull request by c00ler**
Search for Gradle Enterprise agent errors and show a warning if errors are detected:

Screenshot 2023-05-06 at 14 44 23

The feature requires an explicit opt-in from the user:

Screenshot 2023-05-06 at 14 55 53

**TODO**:

- [ ] ~update README~ - not needed, since this functionality is not yet available for the user
- [x] test with new agents when released

### CODE REVIEW

File: src/main/java/hudson/plugins/gradle/GradleLogProcessor.java
Comment (welandaz) [inline src/main/java/hudson/plugins/gradle/GradleLogProcessor.java:1]: I still don't see the value for having this interface. It's only used via `AbstractGradleLogProcessor` and there is currently no use for implementing it separately. I'd move `processLogLine` to `AbstractGradleLogProcessor` as abstract method. 

Do you have some more plans for it? Or perhaps forsee the usage of it elsewhere?

File: src/test/groovy/hudson/plugins/gradle/injection/GradleEnterpriseExceptionLogProcessorTest.groovy [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            scanDetails.isEmpty()
+            !hasMavenErrors
+            hasGradleErrors
+        }
+    }
+
+    def "detects errors for Maven and Gradle"() {
+        when:
+        processor.processLogLine("Internal error in Gradle Enterprise Gradle plugin: com.acme.FooBar")
+        processor.processLogLine("[ERROR] Internal error in Gradle Enterprise Maven extension: com.acme.FooBar")
```
Comment (welandaz) [inline src/test/groovy/hudson/plugins/gradle/injection/GradleEnterpriseExceptionLogProcessorTest.groovy:54]: How about extracting these into constants? They are used in other tests too

**Review by welandaz — APPROVED** [review]
Great work!
Left some small comments

File: src/main/java/hudson/plugins/gradle/GradleLogProcessor.java
Comment (c00ler) [reply src/main/java/hudson/plugins/gradle/GradleLogProcessor.java:1]: Agree. Let's keep it simple. Removed.

**Comment by c00ler** [comment]
@welandaz @alextu For now I've removed the toggle from the UI: https://github.com/jenkinsci/gradle-plugin/pull/294/commits/7a8f660cea860970ede2a3719f77bbfb2c6c89c0 The feature will not be available for the users.

**Review by welandaz — APPROVED** [review]

**Pull request merged**